### ISSUE TITLE
Error when using accounts.js with apollo subscriptions

### ISSUE THREAD

**Issue by nsine** [accounts-js/accounts#843]
When I try to call any subscription in our app which is using @accounts - I'm getting this error which is coming from graphql-modules:
```
TypeError: Cannot use 'in' operator to search for 'session' in undefined
    at Object.normalizeSession (/Users/nsine/Documents/work/mobilehead/sputnik/node_modules/@graphql-modules/core/src/helpers.ts:4:16)
    at GraphQLModule._cache.contextBuilder (/Users/nsine/Documents/work/mobilehead/sputnik/node_modules/@graphql-modules/core/src/graphql-module.ts:1352:19)
    at /Users/nsine/Documents/work/mobilehead/sputnik/node_modules/@graphql-modules/core/src/graphql-module.ts:1363:54
    at Array.map (<anonymous>)
    at /Users/nsine/Documents/work/mobilehead/sputnik/node_modules/@graphql-modules/core/src/graphql-module.ts:1363:33
    at new Promise (<anonymous>)
    at GraphQLModule._cache.contextBuilder (/Users/nsine/Documents/work/mobilehead/sputnik/node_modules/@graphql-modules/core/src/graphql-module.ts:1356:13)
    at Object.context (/Users/nsine/Documents/work/mobilehead/sputnik/packages/gql-server/modules/auth/accounts/index.ts:36:21)
    at Object.<anonymous> (/Users/nsine/Documents/work/mobilehead/sputnik/packages/gql-server/utils/apollo/mergeModules.ts:23:35)
    at step (/Users/nsine/Documents/work/mobilehead/sputnik/packages/gql-server/utils/apollo/mergeModules.ts:44:23)
    at Object.next (/Users/nsine/Documents/work/mobilehead/sputnik/packages/gql-server/utils/apollo/mergeModules.ts:25:53)
    at fulfilled (/Users/nsine/Documents/work/mobilehead/sputnik/packages/gql-server/utils/apollo/mergeModules.ts:16:58)
    at processTicksAndRejections (internal/process/task_queues.js:89:5)
```
https://github.com/Urigo/graphql-modules/blob/master/packages/core/src/helpers.ts#L8

So basically it happens because subscription context builder doesn't expect familiar `req` and `res` params, but `connection` and `payload`. GraphQL modules expect to have a `connection.context.session` but there is no such property in context of our app. So I'd like to check if this could be an issue that can be fixed @accounts-level or not?

As a temporary workaround we are doing something like:
```typescript
context: async (session): Promise<IAccountsContext> => {
    if (!session.req) {
      const reqLike = {
        headers: session.connection.context,
      } as IncomingMessage;

      return accounts.context(reqLike);
    }

    return accounts.context(session);
  },
````

But it would be nice if we can fix this in the package.

@accounts version - 0.19.0

**Comment by flashtheman** [comment]
I get the error as well when I use the graphql playground.

However, when I use the apollo client library, the subscription works partly. The only issue is that the subscription always returns null - I don't know if that's related to accounts-js.

**Comment by jamaldine** [comment]
@nsine thank you. I'd like to know if your solution is added to the package

**Comment by pradel** [comment]
Fixed by #1137
